packages/core-typings/src/IModerationReport.ts

```diff
@@ -11,15 +16,25 @@ export interface IModerationInfo {
 }
 
 export interface IModerationReport extends IRocketChatRecord {
-	message: IMessage;
+	message?: IMessage;
+	room?: Pick<IRoom, '_id' | 'name' | 'fname' | 't' | 'federated' | 'prid'>;
+	reportedUser?: Pick<IUser, '_id' | 'username' | 'name'>;
 	description: string;
 	ts: Date | string;
-	room: Pick<IRoom, '_id' | 'name' | 'fname' | 't' | 'federated' | 'prid'>;
 	reportedBy: Pick<IUser, '_id' | 'username' | 'name' | 'createdAt'>;
 	moderationInfo?: IModerationInfo;
 	_hidden?: boolean;
 }
 
+export type MessageReport = Omit<IModerationReport, 'reportedUser'> & {
+	room: Exclude<IModerationReport['room'], undefined>;
+	message: Exclude<IModerationReport['message'], undefined>;
+};
+
+export type UserReport = Omit<IModerationReport, 'message' | 'room'> & {
+	reportedUser: Exclude<IModerationReport['reportedUser'], undefined>;
```

Comment:
I was talking to @Dnouv and we're wondering if user reports should also include a room :thinking: 
Not sure if it's the right path to have it as a required param or if we should even have it at all. Maybe using as an optional param could work
I think rooms are unrelated enough from user reports to make us not store them, but I'd like to have architecture's opinion on this
cc @sampaiodiego

Reply:
if it is not being used today or really no plans from product either, i don't think it's worth adding an unrelated patch to this pr.

Reply:
agreed